### PR TITLE
pulumi: 3.99.0 -> 3.122.0

### LINKED ISSUE #285848: Pulumi language plugin warning
### Describe the bug
Pulumi does some path comparison to ensure that its language plugin binaries 'match' its main cli -- essentially that they're installed in the same place. The `pulumi` package packages the languages separately and thus, the path comparison fails and pulumi spits out warnings:
```
    warning: using pulumi-language-nodejs from $PATH at /nix/store/v7qgln7y0ag3yq17c2rdpda6r392gr44-devenv-profile/bin/pulumi-language-nodejs
```
```
$ ls -l /nix/store/v7qgln7y0ag3yq17c2rdpda6r392gr44-devenv-profile/bin/pulumi*
lrwxrwxrwx 1 root root  68 Dec 31  1969 /nix/store/v7qgln7y0ag3yq17c2rdpda6r392gr44-devenv-profile/bin/pulumi -> /nix/store/sz30zi87gcb61rqd6dxhqb6kmz131w48-pulumi-3.93.0/bin/pulumi
lrwxrwxrwx 1 root root 100 Dec 31  1969 /nix/store/v7qgln7y0ag3yq17c2rdpda6r392gr44-devenv-profile/bin/pulumi-language-nodejs -> /nix/store/8scsrp9ff1fnyypwmbjb8ll2q4g8yg7x-pulumi-language-nodejs-3.93.0/bin/pulumi-language-nodejs
```

Pulumi bug with some extra discussion:
https://github.com/pulumi/pulumi/issues/14525

I'm a n00b on nix packaging, but would be be possible to have a single `pulumi` package with options to enable the plugins of your choosing?

@trundle
@veehaitch

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.0-91-generic, Ubuntu, 22.04.3 LTS (Jammy Jellyfish), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.16.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by shahinism**
## Description of changes

There are multiple versions skipped until this update containing many changes which can be seen [here](https://github.com/pulumi/pulumi/releases). This update also fixes #285848 issue, by setting `LD_LIBRARY_PATH` based on the work done for `pulumi-bin`. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by brianmay** [comment]
As a new pulumin user, I tested this, seems to work. Although it does give a lot of warnings like:

```
warning: using pulumi-language-go from $PATH at /nix/store/j1kjljhp66ia1ml9ssj47mglm25v09pf-pulumi-language-go-3.122.0/bin/pulumi-language-go
```

**Comment by Mastermindaxe** [comment]
Can't build this on `aarch64-darwin`. Not sure if this is some sort of rate-limiting thing though bc I get timeouts on the checks with 600 seconds and then on a retry I get rate-limited. I'll try and test a bit to see if it's fine or if it's broken. Just a heads-up

**Comment by shahinism** [comment]
@Mastermindaxe could you install the old version? There is a patch to disable some tests which according to the comment should not be necessary after version 3.100.0, but it was, so I kept it untouched, but maybe we need to expand it for mac? 🤔

**Comment by brianmay** [comment]
At one stage I got github rate limit errors, but after the build. Was going to post here but thought it probably not darwin specific. And it stopped happening. But now I am reconsidering. The error was (I still have it on screen):

```
 ❯ pulumi import --from terraform ddd
E0709 11:27:14.169900   88490 log.go:91] GitHub rate limit exceeded for https://api.github.com/repos/pulumi/pulumi-converter-terraform/releases/latest, try again in 7m51.830125s. You can set GITHUB_TOKEN to make an authenticated request with a higher rate limit.
error: load converter plugin: install "terraform": could not find latest version for provider terraform: rate limit exceeded: 403 HTTP error fetching plugin from https://api.github.com/repos/pulumi/pulumi-converter-terraform/releases/latest
```

My thoery was I would be able to avoid these if I added the following to my git config to tell it to use ssh instead of https:

```
[url "ssh://git@github.com/"]
        insteadOf = "https://github.com/"
```

But of course this won't happen if there are rate limit errors happening during the build.

This was also a curious error, but I think this one was 100% my fault (you convert from terraform using the import command not the convert command). But sure as hell was confusing that github responded with 401 not 404 - the actual error was a not found error.

```
$ pulumi convert --from terraform --language go -C ~/tree/ea/infra/terra/roots/sfp/muspelheim/ --out .
Converting from terraform...
Downloading provider: terraform
warning: /Users/brmay/tree/ea/infra/terra/roots/sfp/muspelheim/gitlab.tf:11,1-39: Failed to get provider info; Failed to get provider info for "terraform_remote_state": could not find mapping information for provider terraform; try installing a pulumi plugin that supports this terraform provider
error: <nil>: Error accessing remote module registry; Failed to retrieve available versions for gitlab.ea.com/cloudcell/cidr4/aws: error looking up module versions: 401 Unauthorized
error: conversion failed
```

**Comment by Mastermindaxe** [comment]
> Can't build this on `aarch64-darwin`. Not sure if this is some sort of rate-limiting thing though bc I get timeouts on the checks with 600 seconds and then on a retry I get rate-limited. I'll try and test a bit to see if it's fine or if it's broken. Just a heads-up

Same error. Running into the rate-limiting. I guess I'm too dumb to authenticate myself with a token to Github :sweat_smile: I'll try again later. Should be able to fix that...

**Comment by brianmay** [comment]
Nix builds operate in a pristine environment, I think you won't be able to set `GITHUB_TOKEN` here.

Are you able to post the logs of the failed build?

**Comment by Mastermindaxe** [comment]
I don't think this darwin/personal-specific issue should block this PR going forward. What do you guys think? And @brianmay was able to build on a darwin target so it should be possible I guess :smile:

One of many very similar stack-traces. I think this should be the culprit. Timeout with 600 seconds and `http.send` invocations. I've tried setting a github token via my nix.conf but so far to no avail... Old pulumi version:
```
goroutine 45806 [select]:
net.(*netFD).connect.func2()
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/fd_unix.go:118 +0x70
created by net.(*netFD).connect in goroutine 45859
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/fd_unix.go:117 +0x2b4

goroutine 46005 [IO wait]:
internal/poll.runtime_pollWait(0x128f81448, 0x77)
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/runtime/netpoll.go:345 +0xa0
internal/poll.(*pollDesc).wait(0x14003f04200?, 0x0?, 0x0)
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitWrite(...)
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/internal/poll/fd_poll_runtime.go:93
internal/poll.(*FD).WaitWrite(...)
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/internal/poll/fd_unix.go:683
net.(*netFD).connect(0x14003f04200, {0x101464cd0, 0x14005878000}, {0x1004cce6c?, 0x101e545b8?}, {0x1014557a0?, 0x1400359e000?})
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/fd_unix.go:141 +0x560
net.(*netFD).dial(0x14003f04200, {0x101464cd0, 0x14005878000}, {0x101469ed0?, 0x0?}, {0x101469ed0, 0x14002d20480}, 0x1005dfa9c?)
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/sock_posix.go:124 +0x300
net.socket({0x101464cd0, 0x14005878000}, {0x100f7b8d0, 0x3}, 0x2, 0x1, 0x100f7c1fd?, 0x0, {0x101469ed0, 0x0}, ...)
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/sock_posix.go:70 +0x214
net.internetSocket({0x101464cd0, 0x14005878000}, {0x100f7b8d0, 0x3}, {0x101469ed0, 0x0}, {0x101469ed0, 0x14002d20480}, 0x1, 0x0, ...)
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/ipsock_posix.go:154 +0xa4
net.(*sysDialer).doDialTCPProto(0x14001db0000, {0x101464cd0, 0x14005878000}, 0x0, 0x14002d20480, 0x0)
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/tcpsock_posix.go:85 +0xb0
net.(*sysDialer).doDialTCP(...)
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/tcpsock_posix.go:75
net.(*sysDialer).dialTCP(0x14000ec1498?, {0x101464cd0?, 0x14005878000?}, 0x10071d6c4?, 0x14000ec1458?)
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/tcpsock_posix.go:71 +0x70
net.(*sysDialer).dialSingle(0x14001db0000, {0x101464cd0, 0x14005878000}, {0x10145e100, 0x14002d20480})
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/dial.go:651 +0x1ec
net.(*sysDialer).dialSerial(0x14001db0000, {0x101464cd0, 0x14005878000}, {0x140057f21a0?, 0x1, 0x140057f21a0?})
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/dial.go:616 +0x1a0
net.(*sysDialer).dialParallel(0x140057f2190?, {0x101464cd0?, 0x14005878000?}, {0x140057f21a0?, 0x4?, 0x100f7b8d0?}, {0x0?, 0x140053b0048?>
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/dial.go:517 +0x2a0
net.(*Dialer).DialContext(0x140002d8e00, {0x101464880, 0x101d7a980}, {0x100f7b8d0, 0x3}, {0x140053b0048, 0x12})
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/dial.go:508 +0x53c
net/http.(*Transport).dial(0x140017f2ea0?, {0x101464880?, 0x101d7a980?}, {0x100f7b8d0?, 0x14000c7c678?}, {0x140053b0048?, 0x14000c7cdb0?})
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/http/transport.go:1187 +0xdc
net/http.(*Transport).dialConn(0x101d01880, {0x101464880, 0x101d7a980}, {{}, 0x0, {0x14003ffcd80, 0x5}, {0x140053b0048, 0x12}, 0x0})
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/http/transport.go:1648 +0x61c
net/http.(*Transport).dialConnFor(0x101d01880, 0x14006142630)
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/http/transport.go:1485 +0x90
created by net/http.(*Transport).queueForDial in goroutine 45651
        /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/http/transport.go:1449 +0x380
FAIL    github.com/pulumi/pulumi/pkg/v3/engine/lifecycletest    600.963s
```

New build with `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 325527"`:
[log_pulumi_3.122.0.txt](https://github.com/user-attachments/files/16190522/log_pulumi_3.122.0.txt)

---

@brianmay But I should be able to use the github token in my nix.conf. Emphasis on should 😄

**Comment by brianmay** [comment]
Do we know for certain that this is a Darwin specific issue? Sounds like it could happen anywhere. I am not seeing anything that would suggest this is a Darwin specific issue.

**Comment by Mastermindaxe** [comment]
@brianmay Agreed. But that shouldn't hold up this version update because it's not specific to the new changed code, right? I think we should investigate/fix this in another issue and let this PR proceed

**Comment by Mastermindaxe** [comment]
@NixOS/darwin-maintainers Can sbdy else try and build this please to see if you run into the same issue with rate-limiting as I did?
Cheers!

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 325527` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>python312Packages.pulumi</li>
    <li>python312Packages.pulumi-aws</li>
    <li>python312Packages.pulumi-aws-native</li>
    <li>python312Packages.pulumi-aws-native.dist</li>
    <li>python312Packages.pulumi-aws.dist</li>
    <li>python312Packages.pulumi-azure-native</li>
    <li>python312Packages.pulumi-azure-native.dist</li>
    <li>python312Packages.pulumi-command</li>
    <li>python312Packages.pulumi-command.dist</li>
    <li>python312Packages.pulumi-random</li>
    <li>python312Packages.pulumi-random.dist</li>
    <li>python312Packages.pulumi.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>pulumi</li>
    <li>pulumiPackages.pulumi-language-go</li>
    <li>pulumiPackages.pulumi-language-nodejs</li>
    <li>pulumiPackages.pulumi-language-python</li>
    <li>python311Packages.pulumi</li>
    <li>python311Packages.pulumi-aws</li>
    <li>python311Packages.pulumi-aws.dist</li>
    <li>python311Packages.pulumi.dist</li>
  </ul>
</details>

**Comment by mirkolenz** [comment]
Result of `nixpkgs-review pr 325527` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages failed to build:</summary>
  <ul>
    <li>pulumi</li>
    <li>python311Packages.pulumi</li>
    <li>python311Packages.pulumi-aws</li>
    <li>python311Packages.pulumi-aws.dist</li>
    <li>python311Packages.pulumi.dist</li>
    <li>python312Packages.pulumi</li>
    <li>python312Packages.pulumi-aws</li>
    <li>python312Packages.pulumi-aws-native</li>
    <li>python312Packages.pulumi-aws-native.dist</li>
    <li>python312Packages.pulumi-aws.dist</li>
    <li>python312Packages.pulumi-azure-native</li>
    <li>python312Packages.pulumi-azure-native.dist</li>
    <li>python312Packages.pulumi-command</li>
    <li>python312Packages.pulumi-command.dist</li>
    <li>python312Packages.pulumi-random</li>
    <li>python312Packages.pulumi-random.dist</li>
    <li>python312Packages.pulumi.dist</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>pulumiPackages.pulumi-language-go</li>
    <li>pulumiPackages.pulumi-language-nodejs</li>
    <li>pulumiPackages.pulumi-language-python</li>
  </ul>
</details>

**Comment by gador** [comment]
Result of `nixpkgs-review pr 325527` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>pulumi</li>
  </ul>
</details>
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>pulumiPackages.pulumi-language-go</li>
    <li>pulumiPackages.pulumi-language-nodejs</li>
    <li>pulumiPackages.pulumi-language-python</li>
    <li>python311Packages.pulumi</li>
    <li>python311Packages.pulumi-aws</li>
    <li>python311Packages.pulumi-aws.dist</li>
    <li>python311Packages.pulumi.dist</li>
    <li>python312Packages.pulumi</li>
    <li>python312Packages.pulumi-aws</li>
    <li>python312Packages.pulumi-aws-native</li>
    <li>python312Packages.pulumi-aws-native.dist</li>
    <li>python312Packages.pulumi-aws.dist</li>
    <li>python312Packages.pulumi-azure-native</li>
    <li>python312Packages.pulumi-azure-native.dist</li>
    <li>python312Packages.pulumi-command</li>
    <li>python312Packages.pulumi-command.dist</li>
    <li>python312Packages.pulumi-random</li>
    <li>python312Packages.pulumi-random.dist</li>
    <li>python312Packages.pulumi.dist</li>
  </ul>
</details>

**Comment by gador** [comment]
for me it reproducibly fails in `checkPhase` for `pulumi` with a timeout:
```
pulumi> Running phase: checkPhase
pulumi> evaling implicit 'preCheck' string hook
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/authhelpers     0.314s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/backend 0.538s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/backend/display 0.906s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/backend/display/internal/terminal       0.214s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/backend/filestate       1.059s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/backend/httpstate       12.699s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/backend/httpstate/client        0.382s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/cmd/pulumi      1.728s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/codegen 0.359s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/codegen/cgstrings       0.207s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/codegen/convert 0.324s
pulumi> ?       github.com/pulumi/pulumi/pkg/v3/codegen/dotnet  [no test files]
pulumi> ?       github.com/pulumi/pulumi/pkg/v3/codegen/go      [no test files]
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/codegen/hcl2/model      0.391s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/codegen/hcl2/model/pretty       0.213s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/codegen/hcl2/syntax     0.379s
pulumi> ?       github.com/pulumi/pulumi/pkg/v3/codegen/nodejs  [no test files]
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/codegen/nodejs/tstypes  0.248s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/codegen/pcl     2.546s
pulumi> ?       github.com/pulumi/pulumi/pkg/v3/codegen/python  [no test files]
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/codegen/report  1.372s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/codegen/testing/test    0.610s
pulumi> ok      github.com/pulumi/pulumi/pkg/v3/engine  0.428s
``` 
and then it fails with:
```
pulumi> goroutine 29710 [select]:
pulumi> net.(*netFD).connect.func2()
pulumi>         /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/fd_unix.go:118 +0x70                                                                                                  pulumi> created by net.(*netFD).connect in goroutine 29599                                                                                                                                                   pulumi>         /nix/store/05saqcgidraqmn4z82prsp7rbj9hjmwm-go-1.22.5/share/go/src/net/fd_unix.go:117 +0x2b4
pulumi>
pulumi> goroutine 29844 [chan receive]:
pulumi> github.com/pulumi/pulumi/pkg/v3/engine.newDeployment.func1()                                                                                                                                         pulumi>         /private/tmp/nix-build-pulumi-3.122.0.drv-0/pulumi/pkg/engine/deployment.go:193 +0x40
pulumi> created by github.com/pulumi/pulumi/pkg/v3/engine.newDeployment in goroutine 8806
pulumi>         /private/tmp/nix-build-pulumi-3.122.0.drv-0/pulumi/pkg/engine/deployment.go:192 +0x2ac
pulumi> FAIL    github.com/pulumi/pulumi/pkg/v3/engine/lifecycletest    600.695s
pulumi> FAIL
```

I can build all other packages from `nixpkgs-review` with my oauth token from github, but not _within_  the derivation of `pulumi`.
Maybe we should just deactivate `checkPhase` if it constantly fails due to githubs rate limiting?

**Comment by brianmay** [comment]
I tried building on x86_64-linux, it worked fine. I tried building on x86_64-darwin, it failed to build python3.12-pulumi.

I think this matches up with the results above.

Will try x86_64-darwin again tomorrow.

**Comment by brianmay** [comment]
> I can build all other packages from nixpkgs-review with my oauth token from github, but not within the derivation of pulumi.
Maybe we should just deactivate checkPhase if it constantly fails due to githubs rate limiting?

When i tested x86_64-darwin before and it worked, I probably had not invoked the checkPhase stuff.

**Comment by gador** [comment]
> > I can build all other packages from nixpkgs-review with my oauth token from github, but not within the derivation of pulumi.
> > Maybe we should just deactivate checkPhase if it constantly fails due to githubs rate limiting?
> 
> When i tested x86_64-darwin before and it worked, I probably had not invoked the checkPhase stuff.

This should be invoked automatically without any intervention

**Comment by brianmay** [comment]
The `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 325527"` command appears to have worked fine on my aarch64-linux box (M2 Macbook Pro).

Which is kind of odd and unexpected.

**Comment by Mastermindaxe** [comment]
> The `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 325527"` command appears to have worked fine on my aarch64-linux box (M2 Macbook Pro).
> 
> Which is kind of odd and unexpected.

That is indeeed surprising 😄

---

Nevertheless I'd like to move forward this version update as I don't see the need to block it bc of an already unstable build process on darwin. @Trundle @veehaitch Did you get a chance to review this yet?

---

We should however open another issue for the broken check step on Darwin. Or we should disable the check step (specifically for Darwin) altogether as was suggested further above.

**Review by Mastermindaxe — APPROVED** [review]
LGTM. The darwin build issue should not be part of this PR/code change imo. I'm in favour of moving this along

**Comment by aaronmondal** [comment]
FYI the darwin ratelimiting issue already affected 3.99. We ended up resorting to `doCheck = false;`. The ratelimiting that occurs during the tests blocks the user's IP from using GitHub for an extended amount of time.

**Comment by cnemo-cenic** [comment]
Has anyone tested the withPackages part of this package? It stopped working with this upgrade.

File: pkgs/tools/admin/pulumi/default.nix
Comment (cnemo-cenic) [inline pkgs/tools/admin/pulumi/default.nix:134]: You're missing a quote at the end of this line.

Comment (cnemo-cenic) [reply pkgs/tools/admin/pulumi/default.nix:134]: and why remove the PATH change? isn't that needed for pulumi to actually see the packages?

**Pull request merged**
